In [2]:
import pandas as pd

In [3]:
df = pd.read_csv(r'C:\Users\porte\OneDrive\INTEX\Simplified_PIT_data.csv')

In [4]:
# See unsheltered column names
columnList = df.columns
columnList[20:]

Index(['Unsheltered Homeless', 'Unsheltered Homeless - Under 18',
       'Unsheltered Homeless - Age 18 to 24', 'Unsheltered Homeless - Over 24',
       'Unsheltered Homeless - Female', 'Unsheltered Homeless - Male',
       'Unsheltered Homeless - Transgender',
       'Unsheltered Homeless - Gender that is not Singularly Female or Male',
       'Unsheltered Homeless - Non-Hispanic/Non-Latin(o)(a)(x)',
       'Unsheltered Homeless - Hispanic/Latin(o)(a)(x)',
       'Unsheltered Homeless - White',
       'Unsheltered Homeless - Black, African American, or African',
       'Unsheltered Homeless - Asian or Asian American',
       'Unsheltered Homeless - American Indian, Alaska Native, or Indigenous',
       'Unsheltered Homeless - Native Hawaiian or Other Pacific Islander',
       'Unsheltered Homeless - Multiple Races',
       'Sheltered Total Homeless - Gender Questioning',
       'Unsheltered Homeless - Gender Questioning'],
      dtype='object')

In [5]:
# Convert all columns to numeric data for statistical analysis
numeric_df = df.apply(pd.to_numeric, errors='coerce')


In [6]:
numeric_df.describe()

CoC Number  CoC Name  Count Types         Year  \
count         0.0       0.0          0.0  2311.000000   
mean          NaN       NaN          NaN  2019.501947   
std           NaN       NaN          NaN     1.708869   
min           NaN       NaN          NaN  2017.000000   
25%           NaN       NaN          NaN  2018.000000   
50%           NaN       NaN          NaN  2020.000000   
75%           NaN       NaN          NaN  2021.000000   
max           NaN       NaN          NaN  2022.000000   

       Sheltered Total Homeless  Sheltered Total Homeless - Under 18  \
count               2311.000000                          2311.000000   
mean                 910.771960                           244.853743   
std                 3736.543472                          1214.650357   
min                    1.000000                             0.000000   
25%                  192.500000                            44.000000   
50%                  372.000000                            91.000000   
75%                  800.000000                           182.000000   
max                74982.000000                         25825.000000   

       Sheltered Total Homeless - Age 18 to 24  \
count                              2311.000000   
mean                                 74.235396   
std                                 355.573728   
min                                   0.000000   
25%                                  13.000000   
50%                                  28.000000   
75%                                  62.000000   
max                                7553.000000   

       Sheltered Total Homeless - Over 24  Sheltered Total Homeless - Female  \
count                         2311.000000                        2311.000000   
mean                           591.682821                         401.865859   
std                           2181.750970                        1773.010770   
min                              1.000000                           0.000000   
25%                            129.000000                          85.000000   
50%                            245.000000                         162.000000   
75%                            560.000000                         331.000000   
max                          43193.000000                       36754.000000   

       Sheltered Total Homeless - Male  ...  \
count                      2311.000000  ...   
mean                        504.316313  ...   
std                        1948.181359  ...   
min                           0.000000  ...   
25%                         103.000000  ...   
50%                         206.000000  ...   
75%                         468.500000  ...   
max                       38550.000000  ...   

       Unsheltered Homeless - Non-Hispanic/Non-Latin(o)(a)(x)  \
count                                        2275.000000        
mean                                          370.373187        
std                                          1366.502614        
min                                             0.000000        
25%                                            23.000000        
50%                                            84.000000        
75%                                           306.000000        
max                                         29364.000000        

       Unsheltered Homeless - Hispanic/Latin(o)(a)(x)  \
count                                     2276.000000   
mean                                       113.228032   
std                                        803.829135   
min                                          0.000000   
25%                                          1.000000   
50%                                          6.000000   
75%                                         33.000000   
max                                      21270.000000   

       Unsheltered Homeless - White  \
count                   2276.000000   
mean                     276.792619   
std                     1032.02

In [7]:
numeric_df['Unsheltered Homeless - Female'].isna().sum()

34

In [8]:
# Drop rows with null values in female and male columns
numeric_df.dropna(subset=['Unsheltered Homeless - Female'], inplace=True)
numeric_df.dropna(subset=['Unsheltered Homeless - Male'], inplace=True)


In [9]:
# Pearson r test for gender
from scipy import stats
r, p = stats.pearsonr(numeric_df['Unsheltered Homeless - Female'], numeric_df['Unsheltered Homeless - Male'])
print(f'r: {r}')
print(f'p: {p}')


r: 0.9831776076616021
p: 0.0


The Pearson correlation coefficient r of 0.983 indicates a very strong positive linear relationship between the male and female homelessness. This suggests that as the number of unsheltered homeless females increases, the number of unsheltered homeless males also tends to increase in a very consistent manner.

The p-value of 0.0 means that the correlation is statistically significant, indicating strong evidence against the null hypothesis of no correlation. In other words, there's a very high likelihood that this correlation is not due to random chance.

In [10]:
# Comparing male to female homelessness
male_homeless = numeric_df['Unsheltered Homeless - Male'].sum()
female_homeless = numeric_df['Unsheltered Homeless - Female'].sum()
ratio = male_homeless / female_homeless
print(f'Ratio of male homelessness to female homelessness: {round(ratio,2)}')

Ratio of male homelessness to female homelessness: 2.34


The strong correlation between male and female homelessness does not mean that there is the same rate of male and female homelessness. The ratio of male to female homelessness is 2.34, meaning that for every 100 homeless women, there are 234 homeless men.


In [11]:
# Comparing sheltered to unsheltered homelessness
total_sheltered = numeric_df['Sheltered Total Homeless'].sum()
total_unsheltered = numeric_df['Unsheltered Homeless'].sum()
print(f"Total Sheltered: {total_sheltered}, Total Unsheltered: {total_unsheltered}")
print(f'Ratio of sheltered to unsheltered: {round(total_sheltered/total_unsheltered, 2)}')

Total Sheltered: 2089284, Total Unsheltered: 1101802
Ratio of sheltered to unsheltered: 1.9


In [12]:
# Age and Homelessness anova test
from scipy import stats
f_stat, p_value = stats.f_oneway(
    numeric_df['Sheltered Total Homeless - Under 18'],
    numeric_df['Sheltered Total Homeless - Age 18 to 24'],
    numeric_df['Sheltered Total Homeless - Over 24']
)

print("F-statistic:", f_stat)
print("P-value:", p_value)

F-statistic: 74.58541140655979
P-value: 9.051699738832837e-33


This test shows a statistically significant difference in the means of the three age groups of sheltered homeless individuals. The very small p-value indicates that the number of sheltered homeless individuals significantly differs across the groups of "Under 18," "Age 18 to 24," and "Over 24."

In [13]:
from statsmodels.stats.multicomp import pairwise_tukeyhsd

# Tukey's HSD test
tukey_result = pairwise_tukeyhsd(
    endog=numeric_df[['Sheltered Total Homeless - Under 18', 
                      'Sheltered Total Homeless - Age 18 to 24', 
                      'Sheltered Total Homeless - Over 24']].melt()['value'], 
    groups=numeric_df[['Sheltered Total Homeless - Under 18', 
                       'Sheltered Total Homeless - Age 18 to 24', 
                       'Sheltered Total Homeless - Over 24']].melt()['variable'],
    alpha=0.05
)

print(tukey_result)


                                 Multiple Comparison of Means - Tukey HSD, FWER=0.05                                 
                 group1                                group2               meandiff p-adj    lower    upper   reject
---------------------------------------------------------------------------------------------------------------------
Sheltered Total Homeless - Age 18 to 24  Sheltered Total Homeless - Over 24 521.0316    0.0  419.1307 622.9325   True
Sheltered Total Homeless - Age 18 to 24 Sheltered Total Homeless - Under 18 172.2227 0.0002   70.3217 274.1236   True
     Sheltered Total Homeless - Over 24 Sheltered Total Homeless - Under 18 -348.809    0.0 -450.7099 -246.908   True
---------------------------------------------------------------------------------------------------------------------


A p-value of less than 0.05 in Tukey's HSD (Honestly Significant Difference) test indicates that there is a statistically significant difference between the two groups being compared. Specifically, it suggests that the means of the two groups are different enough that this difference is unlikely to have occurred by random chance.
Since the p-values are all less than 0.05, it means that for each pair of groups tested, there is a statistically significant difference in the number of sheltered homeless individuals between those groups.

In [14]:
# Ethnicity and homelessness anova
ethnicity_groups = numeric_df[['Sheltered Total Homeless - Hispanic/Latin(o)(a)(x)', 'Sheltered Total Homeless - Black, African American, or African', 'Sheltered Total Homeless - White', 'Sheltered Total Homeless - Asian or Asian American',
       'Sheltered Total Homeless - American Indian, Alaska Native, or Indigenous',
       'Sheltered Total Homeless - Native Hawaiian or Other Pacific Islander',
       'Sheltered Total Homeless - Multiple Races', 'Unsheltered Homeless',]].values
f_stat, p_value = stats.f_oneway(ethnicity_groups[0], ethnicity_groups[1], ethnicity_groups[2], ethnicity_groups[3], ethnicity_groups[4], ethnicity_groups[5], ethnicity_groups[6])
print(f"F-statistic: {f_stat}, P-value: {p_value}")


F-statistic: 1.3511326701610447, P-value: 0.2530993315372138


Since the p-value is greater than 0.05, we fail to reject the null hypothesis. This means that there is no statistically significant difference in the number of homeless individuals across different ethnicities. In other words, ethnicity does not appear to have a significant effect on homelessness in your dataset.

In [16]:
# Pearson r test for sheltered vs unsheltered
from scipy import stats
r, p = stats.pearsonr(numeric_df['Unsheltered Homeless'], numeric_df['Sheltered Total Homeless'])
print(f'r: {r}')
print(f'p: {p}')

r: 0.28256175310333964
p: 4.613853341261202e-43


Since the p-value is far below 0.05, we reject the null hypothesis, indicating that there is a significant relationship between being sheltered or unsheltered. This suggests that the number of people who are sheltered versus unsheltered is not random and may depend on factors like resources, access to shelters, policies, or other external factors.